examples/http3-server.rs

```diff
@@ -440,32 +440,42 @@ fn build_response(
 ) -> Result<(std::vec::Vec<quiche::h3::Header>, std::vec::Vec<u8>), ()> {
     let mut file_path = std::path::PathBuf::from(root);
     let mut path = std::path::Path::new("");
+    let mut method = None;
+    let status;
+    let mut body = Vec::new();
 
     for hdr in request {
         match hdr.name() {
             ":path" => {
                 path = std::path::Path::new(hdr.value());
             },
 
-            ":method" =>
-                if hdr.value() != "GET" {
-                    return Err(());
-                },
+            ":method" => method = Some(hdr.value()),
 
             _ => (),
         }
     }
 
-    for c in path.components() {
-        if let std::path::Component::Normal(v) = c {
-            file_path.push(v)
+    if method == Some("GET") {
+        for c in path.components() {
+            if let std::path::Component::Normal(v) = c {
+                file_path.push(v)
+            }
         }
-    }
 
-    let (status, body) = match std::fs::read(file_path.as_path()) {
-        Ok(data) => (200, data),
-        Err(_) => (404, b"Not Found!".to_vec()),
-    };
+        match std::fs::read(file_path.as_path()) {
+            Ok(data) => {
```

Comment:
FTR, from in person chat, this could probably be simplified by using tuples to return (status, data), rather than mutable variables.